izumi-reflect/izumi-reflect/src/test/scala/izumi/reflect/test/SharedLightTypeTagProgressionTest.scala

```diff
@@ -78,6 +78,14 @@ abstract class SharedLightTypeTagProgressionTest extends TagAssertions {
       }
     }
 
+    "progression test: tautological intersections with Any/Object are still preserved in internal structure despite being useless" in {
+      doesntWorkYet {
+        assertDebugSame(LTT[Object with Option[String]], LTT[Option[String]])
+        assertDebugSame(LTT[Any with Option[String]], LTT[Option[String]])
+        assertDebugSame(LTT[AnyRef with Option[String]], LTT[Option[String]])
+      }
+    }
+
```

Comment:
Could this be handled with a filter?

Reply:
Hmm, what do you mean? Filter on string?

Reply:
Inside of the `flattenAnd`, basically filter out `TypeRepr.of[Any/Object/AnyRef]` if the size is greater than 1.